### PR TITLE
Feature: Add prettier & cypress, update package.json

### PR DESCRIPTION
**Pull request by JuraJuki**
- adds bornfight's prettier --> https://github.com/bornfight/prettier as dependency
- adds cypress, cypress-plugin-tab, eslint-plugin-cypress as dependencies
- updates package.json descriptions and scripts

### CODE REVIEW

**Comment by renatoruk** [comment]
https://github.com/bornfight/prettier 

Let's maybe use this one as a dependency, as this is our official config! What do you think?

**Comment by JuraJuki** [comment]
Prettier is already a dependency from the redux-typescript template. I've only added our official config and an ignore file.

**Comment by renatoruk** [comment]
Yes, yes, I see!

But, there are other ways to solve this in a way that's easy to upgrade in the long term. :) 

Of course, prettier config is a trivial example, but it's a good practice to do it consistently for all the dependencies that you can maintain on one place.

What I'm suggesting is listing the dependency in the `package.json`. Maybe it's not visible out of the box in the repository, but the repo actually exposes a [npm package](https://www.npmjs.com/package/@bornfight/prettier) which can be installed on other projects.

Therefore, we can use it as e.g.:
```JSON
{
  "dependencies": {
    "@bornfight/prettier": "^1.0.0.",
  },
  "prettier": "@bornfight/prettier"
}
```

Here is written in detail: https://prettier.io/docs/en/configuration.html#sharing-configurations

**Comment by JuraJuki** [comment]
> Yes, yes, I see!
> 
> But, there are other ways to solve this in a way that's easy to upgrade in the long term. :)
> 
> Of course, prettier config is a trivial example, but it's a good practice to do it consistently for all the dependencies that you can maintain on one place.
> 
> What I'm suggesting is listing the dependency in the `package.json`. Maybe it's not visible out of the box in the repository, but the repo actually exposes a [npm package](https://www.npmjs.com/package/@bornfight/prettier) which can be installed on other projects.
> 
> Therefore, we can use it as e.g.:
> 
> ```json
> {
>   "dependencies": {
>     "@bornfight/prettier": "^1.0.0.",
>   },
>   "prettier": "@bornfight/prettier"
> }
> ```
> 
> Here is written in detail: https://prettier.io/docs/en/configuration.html#sharing-configurations

Oooh that's what you meant! Nice nice, very good...adding!

**Comment by JuraJuki** [comment]
@renatoruk pls check again :)

**Comment by JuraJuki** [comment]
@renatoruk I made a mistake by pushing cypress and package.json update on the same branch. Please review it in this PR. I will update the description :C

File: package.json [hunk 1]
```diff
@@ -54,13 +58,21 @@
...
     "typescript": "~3.8.2"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
+    "format-with-prettier": "prettier --write src/**/*.{ts,tsx}",
+    "cypress-open-integration": "cypress run --spec 'cypress/integration/*'"
   }
```
Comment (renatoruk) [inline package.json:77]: This should be in the `template.json` file. https://github.com/reduxjs/cra-template-redux-typescript/blob/master/template.json

https://create-react-app.dev/docs/custom-templates/#the-templatejson-file

**Review by renatoruk — CHANGES_REQUESTED** [review]

**Comment by renatoruk** [comment]
> @renatoruk I made a mistake by pushing cypress and package.json update on the same branch. Please review it in this PR. I will update the description :C

No worries, it's in the initial phase after all

**Pull request closed without merging**